source/ShipEvent.cpp

```diff
@@ -74,3 +74,25 @@ int ShipEvent::Type() const
 {
 	return type;
 }
+
+
+
+string ShipEvent::TypeToString(int type)
+{
+	static const map<int, string> types = {
```

Comment:
This map does not need to be redefined every function cast.
How about storing it somewhere else?

Reply:
I'm fairly sure it won't be reconstructed/reallocated every time.
It is static and I think this method is static, so it'll either always exist or be constructed on first use.
Putting it in here just scopes it to this method only. If it is used anywhere else, it can be moved to a namespace or the class or something.

Reply:
Indeed static const will only be defined the first time